File: Rentify.Services/Database/Question.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Rentify.Services.Database
{
    public class Question
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [ForeignKey(nameof(UserId))]
        public int UserId { get; set; }
        public User User { get; set; }

        [Required]
        [ForeignKey(nameof(PropertyId))]
        public int PropertyId { get; set; }
        public Property Property { get; set; }

        [Required]
        [MaxLength(1000)]
        public string Content { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public bool IsAnswered { get; set; } = false;

        public Answer? Answer { get; set; }
    }
}

File: UI/rentify_desktop/lib/models/question.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:rentify_desktop/models/answer.dart';
import 'package:rentify_desktop/models/property.dart';
import 'package:rentify_desktop/models/user.dart';


part 'question.g.dart';

@JsonSerializable()
class Question {
  final int id;
  final User? user;
  final Property? property;
  final String content;
  final DateTime createdAt;
  final bool isAnswered;
  final Answer? answer;

  Question({
    required this.id,
    this.user,
    this.property,
    required this.content,
    required this.createdAt,
    required this.isAnswered,
    this.answer
  });

  factory Question.fromJson(Map<String, dynamic> json) =>
      _$QuestionFromJson(json);

  Map<String, dynamic> toJson() => _$QuestionToJson(this);
}